ecpy_pulses/declarations.py

```diff
@@ -0,0 +1,685 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright 2015 by Ecpy Authors, see AUTHORS for more details.
+#
+# Distributed under the terms of the BSD license.
+#
+# The full license is in the file LICENCE, distributed with this software.
+# -----------------------------------------------------------------------------
+"""Enaml objects used to declare sequences in a plugin manifest.
+
+"""
+from __future__ import (division, unicode_literals, print_function,
+                        absolute_import)
+
+
+from traceback import format_exc
+from inspect import cleandoc
+
+from future.utils import python_2_unicode_compatible
+from atom.api import Unicode, Dict, Property
+from enaml.core.api import d_, d_func
+from ecpy.utils.declarator import (Declarator, GroupDeclarator, import_and_get)
+
+from .infos import (SequenceInfos, ConfigInfos, ContextInfos, ShapeInfos)
+
+
+class Sequences(GroupDeclarator):
+    """ GroupDeclarator for Sequences.
+
+    Sequences will be stored according to the group of their parent.
+
+    """
+    pass
+
+
+@python_2_unicode_compatible
+class Sequence(Declarator):
+    """Declarator used to contribute a Sequence
+
+    """
+
+    #: path of the sequence object. Paths should be dot separed and the class
+    #: name preceded by ':'.
+    #: The path to any parent GroupDeclarator will be prepended to it.
+    sequence = d_(Unicode())
+
+    #: Path to the view object associated to the sequence.
+    #: The path of any parent GroupDeclarator will be prepended.
+    view = d_(Unicode())
+
+    #: Metadata associated to the sequence.
+    metadata = d_(Dict())
+
+    #: Id of the sequence computed from the top-level package and the sequence
+    #: name
+    id = Property(cached=True)
+
+    def register(self, collector, traceback):
+        """Collect sequence and view and add infos to the DeclaratorCollector
+        contributions member.
+
+        The group declared by a parent if any is taken into account. All
+        Interface children are also registered.
+
+        """
+        # Build the sequence id by assembling the package name and the class
+        # name.
+        sequence_id = self.id
+
+        # If the sequence only specifies a name update the matching infos.
```

Comment:
This does not make sense for the sequences a priori. Task infos can need to be updated for instrument support we may need to keep this for context however.